src/scenes/Password/Password.tsx

```diff
@@ -13,6 +13,7 @@ import { validatePassword } from 'utils/password';
 
 function Password(_: RouteComponentProps) {
   const [storedPassword, storePassword] = usePersistedState('password', null);
+  const enteredPassword = useRef('');
```

Comment:
Using a ref to store the currently entered password at any given time. This allows `handleEnterGallery` to be a stable function without updating every time the entered password is changed